redash/models.py

```diff
@@ -728,9 +728,9 @@ def to_dict(self):
     def unused(cls, days=7):
         age_threshold = datetime.datetime.now() - datetime.timedelta(days=days)
 
-        unused_results = (db.session.query(QueryResult.id).filter(
-            Query.id == None, QueryResult.retrieved_at < age_threshold)
-            .outerjoin(Query))
```

Comment:
Why stop filtering out the results where the query id isn't set? (that's probably for when a query hasn't been saved yet?)

Reply:
The `outerjoin` joins on `Query.latest_query_data_id`, so this deletes results over the age threshold that aren't referenced by any `Query` rows. We don't want that now that there can be multiple results for a single `Query`.